synapse/python_dependencies.py

```diff
@@ -104,6 +104,13 @@
     # hiredis is not a *strict* dependency, but it makes things much faster.
     # (if it is not installed, we fall back to slow code.)
     "redis": ["txredisapi>=1.4.7", "hiredis"],
+    # We pin black so that our tests don't start failing on new releases.
+    "lint": [
+        "isort<=5.0.3",
+        'black==19.10b0;python_version>"3.5"',
```

Comment:
I think instead of specifying that this shouldn't be installed on py3.5 we should make `ALL_OPTIONAL_DEPENDENCIES` not include `lint` (as linting is a dev thing, not a deployment thing and you should have to explicitly install it). See line 120 below.

We should also be able to use `isort==5.0.3` then.